runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -6940,15 +6942,50 @@ void do_remote_amo_V(int v_len, void** opnd1_v, c_nodeid_t* locale_v,
   post_fma_ct_and_wait(locale_v, &post_desc);
 
 #else // HAVE_GNI_FMA_CHAIN_TRANSACTIONS
+
+  //
+  // This GNI is too old to support chained transactions.  Just do
+  // non-blocking operations instead
+  //
+  gni_post_descriptor_t post_desc_v[MAX_CHAINED_AMO_LEN];
+  atomic_bool post_done_v[MAX_CHAINED_AMO_LEN];
+  int cdi_v[MAX_CHAINED_AMO_LEN];
   int vi;
+
+  for (vi=0; vi<v_len; vi++) {
+    // build up the post descriptor
+    post_desc_v[vi].type            = GNI_POST_AMO;
+    post_desc_v[vi].cq_mode         = GNI_CQMODE_GLOBAL_EVENT;
+    post_desc_v[vi].dlvr_mode       = GNI_DLVMODE_PERFORMANCE;
+    post_desc_v[vi].rdma_mode       = 0;
+    post_desc_v[vi].src_cq_hndl     = 0;
+    post_desc_v[vi].remote_addr     = (uint64_t) (intptr_t) object_v[vi];
+    post_desc_v[vi].remote_mem_hndl = remote_mr_v[vi]->mdh;
+    post_desc_v[vi].length          = size_v[vi];
+    post_desc_v[vi].amo_cmd         = cmd_v[vi];
+    post_desc_v[vi].first_operand   = size_v[vi] == 4 ? *(uint32_t*) opnd1_v[vi]:
+                                                        *(uint64_t*) opnd1_v[vi];
+
+    atomic_init_bool(&post_done_v[vi], false);
+    post_desc_v[vi].post_id = (uint64_t) (intptr_t) &post_done_v[vi];
+
+    // initiate the transaction
+    cdi_v[vi] = post_fma(locale_v[vi], &post_desc_v[vi]);
```

Comment:
You might want to increase the CQ size of the CDs as part of this PR.  Here's the long-ish explanation why:
This will consume a CQ entry per transaction.  The CQ length per comm domain (CD) is fixed, set when the CD is created at init time.  The code that finds a suitable CD for each transaction does check that candidate CDs have space left in their CQs, so to some extent the existing code is resilient against bursts of CQ use caused by doing vectors of NB transactions at a time.  Basically, we'll just use multiple CDs where a series of blocking transactions might have used a single one.  But if many threads happen to do vectors of NB transactions all at once then we might actually run out of total CQ entries, since we only typically have as many CDs as we have worker threads.  If this happened it could cause a live-lock and pause, because the code here doesn't consume any CQ events, and thus free any CQ entries, until it's initiated the entire vector of transactions.  So if we run out of CQ entries across all CDs while all the threads are executing this code, we'll hang when since no thread will be able to find a CD with free CQ entries to post on.